applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py

```diff
@@ -0,0 +1,206 @@
+        # Get arguments from JSON file
+        #application         = self.settings["solver_wrapper_settings"]["application"].GetString()
+        working_directory   = self.settings["solver_wrapper_settings"]["working_directory"].GetString()
+        problem_directory   = self.settings["solver_wrapper_settings"]["problem_directory"].GetString()
+        #region              = self.settings["solver_wrapper_settings"]["region"].GetString()
+        echo_level          = self.settings["solver_wrapper_settings"]["echo_level"].GetInt()
+        problem             = self.settings["solver_wrapper_settings"]["problem"].GetString()
+        inputFile           = self.settings["solver_wrapper_settings"]["input_file"].GetString()
+        np                  = self.settings["solver_wrapper_settings"]["num_processors"].GetInt()
+        nt                  = self.settings["solver_wrapper_settings"]["num_threads"].GetInt()
+        gpu                 = self.settings["solver_wrapper_settings"]["gpu_flag"].GetString()
+        rst                 = self.settings["solver_wrapper_settings"]["restart_flag"].GetString()
+        frst                = self.settings["solver_wrapper_settings"]["restart_flag"].GetString()
+        self.deltaT         = self.settings["solver_wrapper_settings"]["time_increment"].GetDouble()
+
+        # ---------------------
+        # Launch AcuSolve
+        # ---------------------
+        common_cmd = inputFile + " -pb " + problem +" -dir "+ working_directory + " -pdir " + problem_directory + " -np "+ str(np) + " -nt " + str(nt)
+        verbose_cmd = " -verbose " + str(echo_level)
+        if platform == "linux" or platform == "linux2":
+            base_cmd = "acuRun -inp "
+            buffer_cmd = " &"
+            if gpu != "FALSE":
+                cmd = base_cmd + common_cmd + " -gpu " + gpu + verbose_cmd + buffer_cmd  
+            elif rst != "FALSE":
+                cmd = base_cmd + common_cmd + " -rst " + verbose_cmd + buffer_cmd  
+            elif frst != "FALSE":
+                cmd = base_cmd + common_cmd + " -frst " + verbose_cmd + buffer_cmd  
+            else:
+                cmd = base_cmd + common_cmd + verbose_cmd + buffer_cmd
+        elif platform == "win32":
+            base_cmd = "acuRun.bat -inp "
+            buffer_cmd = " -lbuff"
+            if gpu != "FALSE":
+                cmd = base_cmd + common_cmd + " -gpu " + gpu + verbose_cmd + buffer_cmd  
+            elif rst != "FALSE":
+                cmd = base_cmd + common_cmd + " -rst " + verbose_cmd + buffer_cmd   
+            elif frst != "FALSE":
+                cmd = base_cmd + common_cmd + " -frst " + verbose_cmd + buffer_cmd
+            else:
+                cmd = base_cmd + common_cmd + verbose_cmd + buffer_cmd
+        else:
+            # Python exception in case not detected OS
+            raise Exception("Unsupported operating system detected.")
+        cs_tools.cs_print_info(self.name + ": " +  cmd)
+        os.system(cmd)
```

Comment:
That is even worse 😅 (talking mainly about security issues, had sth similar at work)

I recommend using subprocess.run